backend/open_webui/internal/pocketbase/ROLLOUT_AND_MIGRATION.md

```diff
@@ -0,0 +1,36 @@
+# Migration Plan (One-Shot)
+
+This document outlines a one-shot migration from SQLAlchemy storage to PocketBase (PB) for Open WebUI, along with validation and backout steps. No staged/feature-flag rollout in production.
+
+## Preparation
+- [ ] Stand up PB; configure admin/service credentials.
+- [ ] Copy JS migrations to PB `pb_migrations/`; copy hooks to `pb_hooks/`.
+- [ ] Restart PB and verify schema via Admin UI/CLI.
+
+## Adapter Implementation
+- [ ] Implement PB adapters for all domain models (users, chats, messages, tags, files, groups, folders, models, knowledge, notes, channels, feedbacks, prompts, memories).
+- [ ] Update `backend/open_webui/models/*.py` internals to call PB adapters; keep public signatures consistent.
+- [ ] Implement realtime bridge and files integration.
+
+## Data Export & Import
+- [ ] Export SQL data per table to NDJSON/CSV.
+- [ ] Import into PB via Admin UI/CLI or batch scripts using the Python client.
+- [ ] For tags: compute `id_comp = "{id_normalized}:{user_id}"` and create `chats.tags` relations.
+- [ ] For files: upload file binaries and set metadata; verify `hash`.
```

Comment:
Should probably add an option if "source" pg/sqlite-backed open-webui uses local file storage for files, pocketbase can just point to that dir (if also configured with local file storage); if both use object storage like S3, likewise, only files metadata (filenames/uris, hash, etc.) migration between is needed. Don't need to think this through in very much detail yet, just chalk up